src/FileList/FileListHandler.cc

```diff
@@ -137,11 +138,11 @@ void FileListHandler::GetFileList(CARTA::FileListResponse& file_list, std::strin
                     bool is_region_file(false);
 
                     if (region_list && cc_file.isRegular(true)) {
-                        CARTA::FileType file_type(GetRegionType(full_path)); // CRTF, DS9, or UNKNOWN
+                        auto file_type = GuessRegionType(full_path, filter_mode == CARTA::Content);
```

Comment:
I would think from the user comment in carta issue 1 ("I would greatly prefer if we just saw the filenames and sizes (simple ls -lh output) rather than the metadata") and the CARTA::AllFiles mode that you would skip all type guessing, else isn't this actually by extension?  In AllFiles mode, the frontend should leave the file browser Type column blank or hide it if possible rather than list all files as Unknown.

Reply:
@kswang1029 what do you think? I'm assuming that guessing the file type by extension is trivially fast (just some string comparisons), so I can't imagine skipping this process for "All files" would make any difference. If we keep it in, users can at least see which files CARTA _thinks_ it can support.

Reply:
I've just checked, and the performance difference is negligible, so I can't see any practical reason why we shouldn't guess the file type by extension for "All files".

See tests for 25 000 FITS files below. Disk caches cleared between each test.

```shell
# listing 25 000 files from an SSD, filtering by content
[2021-11-15 11:08:17.873] [debug] [performance] File list for directory /home/angus/cubes/fits/lots_dummy filled in 3038.569 ms at 8227.557 entries/s

# listing 25 000 files from an SSD, filtering by extension
[2021-11-15 11:08:37.931] [debug] [performance] File list for directory /home/angus/cubes/fits/lots_dummy filled in 355.259 ms at 70371.194 entries/s

# listing 25 000 files from an SSD, showing all files and not checking extension
[2021-11-15 11:08:48.008] [debug] [performance] File list for directory /home/angus/cubes/fits/lots_dummy filled in 347.240 ms at 71996.314 entries/s


# listing 25 000 files from an HDD, filtering by content
[2021-11-15 11:13:34.370] [debug] [performance] File list for directory /media/angus/ImageData/dummy filled in 123947.127 ms at 201.699 entries/s

# listing 25 000 files from an HDD, filtering by extension
[2021-11-15 11:13:52.568] [debug] [performance] File list for directory /media/angus/ImageData/dummy filled in 364.601 ms at 68568.106 entries/s

# listing 25 000 files from an HDD, showing all files and not checking extension
[2021-11-15 11:14:04.578] [debug] [performance] File list for directory /media/angus/ImageData/dummy filled in 342.513 ms at 72989.930 entries/s
```

Reply:
Good to guess file type by file extension if the performance impact is negligible. Question here: by extension does this mean only fits and hdf5 images can be guessed? casa and miriad will be listed as folder?

Reply:
> Question here: by extension does this mean only fits and hdf5 images can be guessed? casa and miriad will be listed as folder?

No, they are handled the same way as before. This PR only affects how regular files are handled. I guess if you had thousands of CASA images, this might be a problem? We may need to implement a faster image check for those image types. In particular, something that doesn't involve actually _opening_ the files within the CASA image folder.

Reply:
I was trying to understand the difference between "All files" and "By extension", which seemed to do the same thing.  It appears that Unknown files are listed for the first but not the second.